python/GafferSceneUI/SceneViewUI.py

```diff
@@ -651,20 +657,28 @@ def hasLabel( self ) :
 	def __menuDefinition( self ) :
 
 		expandAll = bool( self.getPlug().getValue() )
+		expandUsdInstancers = self.getPlug().node()["expandUsdInstancers"].getValue()
 
 		m = IECore.MenuDefinition()
 		m.append( "/Expand Selection", { "command" : self.getPlug().node().expandSelection, "active" : not expandAll, "shortCut" : "Down" } )
 		m.append( "/Expand Selection Fully", { "command" : functools.partial( self.getPlug().node().expandSelection, depth = 999 ), "active" : not expandAll, "shortCut" : "Shift+Down" } )
 		m.append( "/Collapse Selection", { "command" : self.getPlug().node().collapseSelection, "active" : not expandAll, "shortCut" : "Up" } )
 		m.append( "/Expand All Divider", { "divider" : True } )
 		m.append( "/Expand All", { "checkBox" : expandAll, "command" : Gaffer.WeakMethod( self.__toggleMinimumExpansionDepth ) } )
+		m.append( "/USD Instancers Divider", { "divider" : True } )
+		m.append( "/Expand USD Instancers When Rendering", { "checkBox" : expandUsdInstancers, "command" : Gaffer.WeakMethod( self.__toggleExpandUsdInstancers ) } )
```

Comment:
This feels kindof wordy. I guess you're trying to get across the difference between using OpenGL and a renderer like Arnold? Is it worth having a separate option for each - seems like it might be useful to expand in OpenGL sometimes (but it would default off)?

Reply:
Happy to do whatever you think is best. Something like a top level menu "Expand USD Instancers" with child checkboxes menu options "OpenGL" and "CPU Renderers" or something? Not sure what to call the latter option - "Not OpenGL" is ugly, but it's probably unnecessary to have separate checkboxes for every renderer we support?

Note that adding an OpenGL option here makes the interaction I mentioned before between Hierarchy Expansion and locations that don't exist in the original scene even more of a problem - OpenGL doesn't support capsules, and in a production scene, you're usually not going to want to have "Expand All" on in OpenGL. I suspect users will be confused when they turn on "Expand USD Instancers > OpenGL", and don't see any instances, because "Expand All" isn't on.

Reply:
As per conversation, this now shows in the UI as just "Expand USD Instancers", but there is a different value for each renderer. This is implemented as an AtomicCompoundData plug on the adaptor with a BoolData per renderer ... feels like a bit of a weird interface, but it's fully internal.